### PR TITLE
Add Android Assist VAD timing settings

### PR DESCRIPTION
**Pull request by lextiz**
## Summary
Adds optional Assist silence and timeout overrides for voice runs. Related: home-assistant/core#122177.

## Checklist
- [x] New or updated tests have been added to cover the changes following the testing guidelines.
- [x] The code follows the project's code style and best practices.
- [ ] The changes have been thoroughly tested, and edge cases have been considered.
- [x] Changes are backward compatible whenever feasible.

### CODE REVIEW

**Comment by lextiz** [comment]
done

File: app/src/main/kotlin/io/homeassistant/companion/android/settings/assist/AssistSettingsViewModel.kt [hunk 1]
```diff
@@ -148,4 +153,33 @@ class AssistSettingsViewModel @Inject internal constructor(
...
+        viewModelScope.launch {
+            assistConfigManager.setVadTimeoutSeconds(input.toPositiveSecondsOrNull())
+        }
+    }
+}
+
+private fun String.sanitizedSecondsInput(): String? {
+    if (isEmpty()) return this
+    if (count { it == '.' } > 1) return null
+    return takeIf { it.all { char -> char.isDigit() || char == '.' } }
```
[the remark this answers is not in the document]
Comment (lextiz) [reply app/src/main/kotlin/io/homeassistant/companion/android/settings/assist/AssistSettingsViewModel.kt]: done

File: common/src/main/kotlin/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt [hunk 2]
```diff
@@ -404,6 +406,29 @@ internal class PrefsRepositoryImpl @Inject constructor(
...
+        setAssistVadSeconds(PREF_ASSIST_VAD_TIMEOUT_SECONDS, seconds)
+    }
+
+    private suspend fun setAssistVadSeconds(key: String, seconds: Double?) {
+        if (seconds == null || seconds <= 0.0) {
+            localStorage().remove(key)
+        } else {
+            localStorage().putString(key, seconds.toString())
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lextiz) [reply common/src/main/kotlin/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt:430]: done

File: common/src/main/kotlin/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt [hunk 3]
```diff
@@ -419,3 +444,5 @@ internal class PrefsRepositoryImpl @Inject constructor(
         localStorage().remove(PREF_ALLOWED_TAGS)
     }
 }
+
+private fun String?.toPositiveDoubleOrNull(): Double? = this?.toDoubleOrNull()?.takeIf { it > 0.0 }
```
[the remark this answers is not in the document]
Comment (lextiz) [reply common/src/main/kotlin/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt]: done

File: common/src/main/kotlin/io/homeassistant/companion/android/common/data/websocket/WebSocketRepository.kt [hunk 4]
```diff
@@ -127,6 +127,8 @@ interface WebSocketRepository {
         pipelineId: String? = null,
         conversationId: String? = null,
         wakeWordPhrase: String? = null,
```
[the remark this answers is not in the document]
Comment (lextiz) [reply common/src/main/kotlin/io/homeassistant/companion/android/common/data/websocket/WebSocketRepository.kt:130]: done

**Comment by lextiz** [comment]
Superseded by #6895 after CLA signing.
